sqlglot/dialects/mysql.py

```diff
@@ -599,6 +610,13 @@ def jsonarraycontains_sql(self, expression: exp.JSONArrayContains) -> str:
             return f"{self.sql(expression, 'this')} MEMBER OF({self.sql(expression, 'expression')})"
 
         def cast_sql(self, expression: exp.Cast, safe_prefix: t.Optional[str] = None) -> str:
+            if expression.to.this in {
+                exp.DataType.Type.TIMESTAMP,
+                exp.DataType.Type.TIMESTAMPTZ,
+                exp.DataType.Type.TIMESTAMPLTZ,
```

Comment:
See comment in type mapping w.r.t to drying this section up.

Reply:
use the Temporal_types constant

Reply:
Temporal includes date

Reply:
That covers all temporal types though right, e.g. `TIME`, `DATETIME`, .. This might be valid, but I'm not 100% sure about MySQL's specifics here; if we only care about `TIMESTAMP[TZ|LTZ]` though, we can just do

```python
if expression.to.this in self.TIMESTAMP_TYPE_MAPPING:
```

Reply:
yea, just use that constant than, otherwise what's the point of it?

Reply:
Excluded DATETIME and then also added Doris support which doesn't seem to have a `TIMESTAMP` func so this changed a bit.